compiler/data/var-data.h

```diff
@@ -44,7 +44,10 @@ class VarData {
   bool marked_as_const = false;
   bool is_read_only = true;
   bool is_foreach_reference = false;
-  int dependency_level = 0;
+  bool is_builtin_runtime = false;        // $_SERVER, $argv, etc., see PhpScriptBuiltInSuperGlobals in runtime
+  int dependency_level = 0;               // for constants only (c_str$, c_arr$, etc)
```

Comment:
Why do we need to store `dependency_level` here?
Why can't we just calculate it once in codegen?

For now this class becomes very overloaded semantically

Reply:
It's calculated earlier, in collect const vars. Codegen uses already calculated values.
Yes, this class if quite complicated, but all these fields are really needed for variable description.